src/cfi.rs

```diff
@@ -63,6 +63,78 @@ impl<'input, Endian> DebugFrame<'input, Endian>
         let (_, entry) = try!(CommonInformationEntry::parse(input));
         Ok(entry)
     }
+
+    /// Find the frame unwind information for the given address.
+    ///
+    /// If found, the unwind information is returned along with the reset
+    /// context in the form `Ok((unwind_info, context))`. If not found,
+    /// `Err(gimli::Error::NoUnwindInfoForAddress)` is returned. If parsing or
+    /// CFI evaluation fails, the error is returned.
+    ///
+    /// ```
+    /// use gimli::{DebugFrame, NativeEndian, UninitializedUnwindContext};
+    ///
+    /// # fn foo() -> gimli::Result<()> {
+    /// # let read_debug_frame_section = || unimplemented!();
+    /// // Get the `.debug_frame` section from the object file.
+    /// let debug_frame = DebugFrame::<NativeEndian>::new(read_debug_frame_section());
+    ///
+    /// # let get_frame_pc = || unimplemented!();
+    /// // Get the address of the PC for a frame you'd like to unwind.
+    /// let address = get_frame_pc();
+    ///
+    /// // This context is reusable, which cuts down on heap allocations.
+    /// let ctx = UninitializedUnwindContext::new();
+    ///
+    /// let (unwind_info, ctx) = try!(debug_frame.unwind_info_for_address(ctx, address));
```

Comment:
It wouldn't because uninitialized -> initialized is consuming `self`, so we can't have a reference.

Agreed regarding `try!` -- we also probably don't need actual tests in the doc examples like how a lot of the unit stuff has (it was convenient at the time) because there's so much noise and we really just want to show how to use the APIs.